packages/vm/src/static_analysis.rs

```diff
@@ -21,26 +21,62 @@ pub fn deserialize_wasm(wasm_code: &[u8]) -> VmResult<Module> {
     })
 }
 
-pub fn exported_functions(module: &Module) -> HashSet<String> {
-    module
-        .export_section()
-        .map_or(HashSet::default(), |export_section| {
-            export_section
-                .entries()
-                .iter()
-                .filter_map(|entry| match entry.internal() {
-                    Internal::Function(_) => Some(entry.field().to_string()),
-                    _ => None,
-                })
-                .collect()
-        })
+/// A trait that allows accessing shared functionality of `parity_wasm::elements::Module`
+/// and `wasmer::Module` in a shared fashion.
+pub trait ExportInfo {
+    /// Returns all exported function names with the given prefix
+    fn exported_function_names(&self, prefix: Option<&str>) -> HashSet<String>;
+}
+
+impl ExportInfo for Module {
+    fn exported_function_names(&self, prefix: Option<&str>) -> HashSet<String> {
+        self.export_section()
+            .map_or(HashSet::default(), |export_section| {
+                export_section
+                    .entries()
+                    .iter()
+                    .filter_map(|entry| match entry.internal() {
+                        Internal::Function(_) => Some(entry.field()),
+                        _ => None,
+                    })
+                    .filter(|name| {
+                        if let Some(required_prefix) = prefix {
+                            name.starts_with(required_prefix)
+                        } else {
+                            true
+                        }
+                    })
+                    .map(|name| name.to_string())
+                    .collect()
+            })
+    }
+}
+
+impl ExportInfo for wasmer::Module {
+    fn exported_function_names(&self, prefix: Option<&str>) -> HashSet<String> {
```

Comment:
I was wondering where this was coming from... I see you used the trait to unify the two types. Nice